src/asciidoctor-kroki.js

```diff
@@ -48,7 +52,7 @@ const processKroki = (processor, parent, attrs, diagramType, diagramText, contex
     diagramText = require('./preprocess.js').preprocessPlantUML(diagramText, context, doc.getBaseDir())
   }
   const blockId = attrs.id
-  const format = attrs.format || 'svg'
+  const format = attrs.format || parent.document.getAttribute('kroki-default-format') || 'svg'
```

Comment:
Asciidoctor (and by extension Asciidoctor.js) provides a function to resolve and attribute:

https://github.com/asciidoctor/asciidoctor.js/blob/2b006467bd99b2ce8589b107e8fc34a6a4323880/packages/core/src/asciidoctor-core-api.js#L903-L906

But I'm fine with the `||`.

Reply:
I think the useful form of that function, with the fallback to the alternate name, is only available in asciidoctor 2.  In order to avoid potential confusion about which version of the function is being used I'd like to stick with the ||.